Zend/zend_builtin_functions.c

```diff
@@ -1061,6 +1061,53 @@ ZEND_FUNCTION(trait_exists)
 }
 /* }}} */
 
+ZEND_FUNCTION(is_enum)
+{
+	// FIXME: Avoid code duplication
+	zval *name_or_object;
+	zend_string *name;
+	zend_string *lcname;
+	zend_class_entry *ce;
+	zend_bool autoload = 1;
+	int flags = ZEND_ACC_ENUM;
+	int skip_flags = 0;
+
+	ZEND_PARSE_PARAMETERS_START(1, 2)
+		Z_PARAM_ZVAL(name_or_object)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_BOOL(autoload)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (Z_TYPE_P(name_or_object) == IS_OBJECT) {
+		name = Z_OBJCE_P(name_or_object)->name;
+	} else if (Z_TYPE_P(name_or_object) == IS_STRING) {
+		name = Z_STR_P(name_or_object);
+	} else {
+		RETURN_FALSE;
+	}
+
+	if (!autoload) {
+		if (ZSTR_VAL(name)[0] == '\\') {
+			/* Ignore leading "\" */
+			lcname = zend_string_alloc(ZSTR_LEN(name) - 1, 0);
+			zend_str_tolower_copy(ZSTR_VAL(lcname), ZSTR_VAL(name) + 1, ZSTR_LEN(name) - 1);
+		} else {
+			lcname = zend_string_tolower(name);
+		}
+
+		ce = zend_hash_find_ptr(EG(class_table), lcname);
+		zend_string_release_ex(lcname, 0);
```

Comment:
```suggestion
		zend_string_release(lcname);
```

`lcname = zend_string_tolower(name);` seems like it might return a persistent string if the class to autoload is already lowercase?